lib/Util.ts

```diff
@@ -18,8 +19,13 @@ export function getUrl(requestUrl: string, baseUrl?: string): string  {
     combined.protocol = combined.protocol || base.protocol;
     combined.auth = combined.auth || base.auth;
     combined.host = combined.host || base.host;
-    // path from requestUrl always wins
 
+    if (preservePath) {
+        combined.pathname = base.pathname + combined.pathname;
```

Comment:
path.join?  I think there is also interesting test cases with base not / having trailing slash and pathname not/having leading slash.

Reply:
Using `path.join` now. Also fixes the trailing and leading slash problems